utils/server.ts

```diff
@@ -15,18 +15,60 @@ const getAuthorizationHeader = async () => {
   }
 };
 
+const objectToFormData = (
+  obj: Record<string, any>,
+  formData: FormData = new FormData(),
+  parentKey: string = ''
+): FormData =>
+  Object.entries(obj).reduce((fd: FormData, [key, value]: [string, any]) => {
+    const propName: string = parentKey ? `${parentKey}[${key}]` : key;
+
+    if (typeof value === 'object' && !(value instanceof File)) {
+      return objectToFormData(value, fd, propName);
+    } else if (Array.isArray(value)) {
+      value.forEach((item: any, index: number) =>
+        objectToFormData(item, fd, `${propName}[${index}]`)
+      );
+    } else {
+      if (value !== undefined) fd.append(propName, value);
```

Comment:
`if (value)`?

Reply:
Value can also  be `null`, and we might want to send that to server, otherwise pattern matching issues may occur.